Sources/ApolloWebSocket/WebSocketTransport.swift

```diff
@@ -89,19 +72,35 @@ public class WebSocketTransport: NetworkTransport, WebSocketDelegate {
 
   private func processMessage(socket: WebSocketClient, text: String) {
     OperationMessage(serialized: text).parse { (type, id, payload, error) in
-      guard let type = type, let messageType = OperationMessage.Types(rawValue: type) else {
-        notifyErrorAllHandlers(WebSocketError(payload: payload, error: error, kind: .unprocessedMessage(text)))
-        return
+      guard
+        let type = type,
+        let messageType = OperationMessage.Types(rawValue: type) else {
+          self.notifyErrorAllHandlers(WebSocketError(payload: payload, error: error, kind: .unprocessedMessage(text)))
+          return
       }
 
-      switch(messageType) {
-      case .data, .error:
-        if let id = id, let responseHandler = subscribers[id] {
-          responseHandler(payload,error)
+      switch messageType {
+      case .data,
+           .error:
+        if
+          let id = id,
+          let responseHandler = subscribers[id] {
+          if let payload = payload {
+            responseHandler(.success(payload))
+          } else if let error = error {
+            responseHandler(.failure(error))
+          } else {
+            let websocketError = WebSocketError(payload: payload,
+                                                error: error,
+                                                kind: .neitherErrorNorPayloadReceived)
+            responseHandler(.failure(websocketError))
```

Comment:
This is a new error type to help handle cases where we previously could have sent `nil` for both the payload and the error.

Reply:
Yeah I plan to look at that in more depth when I do some much more detailed looks at the WebSocket stuff-  this is at least to get this out the door in a way that still gives us some kind of semi-meaningful error about what went wrong.